### PR TITLE
Add Bioelectric Organ mutation

### PR DESCRIPTION
**Pull request by souricelle**
SUMMARY: Content "Add a post-thresh 'electric eel' mutation for Fish"

Compared to many other mutations, Fish seems a little lacking, especially given is heavy downsides. We now have the enchantment system which can expand the functionality of mutations and I immediately thought of electric eels. Electric eels are actually fish (most closely related to catfish) who have in several different cases convergently evolved an ability to stun their prey and predators.

Most (all?) fish have electroreceptors which use electric currents to sense movement in the water around them. Electric eels have simply turned up the power on theirs, and some are capable of producing 860 volts (https://phys.org/news/2019-09-species-electric-eel-highest-voltage.html). The amperage is usually quite low at about 1 amp so this is not likely to be fatal to something the size of a human. It's usually likened to a taser or an electric fence. A survivor is much larger than an electric eel but I decided to keep it in the taser range.

#### Describe the solution

I added in a post-threshold mutation with Electroreceptors as its prereq which uses the enchantment system to proc a zap effect when the user performs a melee attack. The language isn't perfectly clear but the mutant is probably making physical contact with the enemy rather than discharging the effect through their weapon. On average this goes off every 35 attacks, but it can be more or less as the system is a bit random. Damage is 1-2 electric with an additional zap effect.

#### Describe alternatives you've considered

The mutation doesn't check for the user's gear, but that's intentional. Apparently normal clothing isn't enough to stop the shock and the victim would need to have insulated gear to protect themselves. I considered requiring that the mutant be in water to use this ability, but I discovered it's actually a bit stronger out of water (https://www.nationalgeographic.com/news/2016/06/electric-eels-animals-science-shock/#:~:text=In%20an%20unusual%20discovery%2C%20electric,the%20target%20to%20discharge%20shocks.)

I also considered making this a defensive ability, but it seems to be something the eels do actively to hunt prey, so I felt this was better.

#### Testing

I spawned a survivor and had him inject fish serum until he gained the mutation. Then I spawned a flesh wall and had him attack it barehanded, dealing no damage except when his Bioelectric Organ went off. From there I was able to determine that it was proccing as often as I intended and functioning properly.

### CODE REVIEW

**Comment by MutaMan** [comment]
Are you aware that using a "hit_you_effect" causes the thing to trigger even if you miss? It's not a fault of yours though, just a bug yet to be fixed.

File: data/json/mutations/mutation_spells.json [hunk 1]
```diff
@@ -18,5 +18,25 @@
     "flags": [ "SILENT", "NO_LEGS", "NO_HANDS", "RANDOM_DAMAGE", "RANDOM_DURATION", "RANDOM_TARGET" ],
     "energy_source": "NONE",
     "damage_type": "bio"
```
Comment (mqrause) [inline data/json/mutations/mutation_spells.json:20]: Looks like you made this on an older version of the file. Changing the damage type here to "biological" should fix the merge conflict.

Comment (souricelle) [reply data/json/mutations/mutation_spells.json:20]: Bio is for poison effects. It needs to be electric because it's an electric shock.

Comment (souricelle) [reply data/json/mutations/mutation_spells.json]: Ah, I'll update MAGIC.md to reflect that.

**Comment by mqrause** [comment]
Oh wow, it's still considered a merge conflict. Sorry about that, just ignore me.

**Comment by Fris0uman** [comment]
Cab you make it consume calories, and maybe have a cooldown on it?

**Comment by souricelle** [comment]
@Fris0uman Best I can do is change it to an active mutation that drains calories while it's on. I thought the 1/35 proc rate worked well enough that we wouldn't need a cooldown (the animals seem to use the ability willy nilly in the wild until they get tired and have to recharge) and apparently when hunger got changed over to calories, nobody updated mutations, so there's no way to make this or any active mutation turn off automatically when the mutant is low on calories (except when they're starving, but even that doesn't seem to work properly). The enchantment system has a built in cost system but apparently it doesn't apply to on-hit effects from mutations. Note that this means the cephalopod ink ability is also totally free.

The other option is to make it a hardcoded projectile mutation like the quill mutation,, but then the character would be shooting out a shock cloud which isn't how electric eels work.

Also I spotted a typo elsewhere in the file and fixed it (preys to prey).

**Comment by paulenka-aleh** [comment]
Well, does it work while wearing power armor?..

**Pull request closed without merging**